e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js

```diff
@@ -279,41 +279,48 @@ describe(
 
       queryBuilderHeader().within(() => {
         cy.findByLabelText("Back to Sleep dashboard").click();
-        cy.get("@dashboard.all").should("have.length", 1);
-        cy.get("@dashcardQuery.all").should("have.length", 2);
       });
 
       getDashboardCard().within(() => {
         cy.findByText("Sleep card").should("be.visible");
       });
+
+      // dashboard is taken from the cache, no re-fetch
+      cy.get("@dashboard.all").should("have.length", 1);
+      // the query is triggered second time as first one never loaded - no value in the cache
+      cy.get("@dashcardQuery.all").should("have.length", 2);
     });
 
     it("should preserve filter value when navigating between the dashboard and the question", () => {
+      // could be a regular dashboard with card and filters
       createDashboardWithSlowCard();
       cy.get("@dashboardId").then(visitDashboard);
+      cy.wait("@dashcardQuery");
 
-      cy.get("@dashcardQuery.all").should("have.length", 1);
+      // initial loading of the dashboard with card
+      cy.get("@dashcardQuery.all").should("have.length", 2);
 
-      filterWidget().findByPlaceholderText("sleep").type("5{enter}");
+      filterWidget().findByPlaceholderText("sleep").type("1{enter}");
 
-      cy.get("@dashcardQuery.all").should("have.length", 2);
+      // we applied filter, so the data is requested again
+      cy.get("@dashcardQuery.all").should("have.length", 3);
```

Comment:
Unless you explicitly wait for the response, you can never be sure that visiting the question will NOT interrupt and cancel this dashCardQuery request.

Reply:
we explicitly cancel some requests (afaik it can be done to unblock BE when it makes a heavy query), so I expected from cypress it catches all the fired http requests and it's what I wanted to test